### DISCUSSION TITLE
Adding duration to the Notifications component

### DISCUSSION THREAD

**Discussion by aaronstezycki** [alpinejs/alpine#3392] [1. Help] [answered]
So I've added a few lines to try to add 'duration' to the component, 

**parent x-data:** 
```js
add(e) {
    this.notifications.push({
        id: e.timeStamp,
        type: e.detail.type,
        content: e.detail.content,
        duration: e.detail.duration
    })
},
...
```

**Notification x-data:**
```js
init() {
    this.$nextTick(() => this.show = true)
    setTimeout(() => this.transitionOut(), this.duration) // I think this is the problem line
},
```

**Form & Duration Input:**
```html
<form 
    x-data="{
        content: 'Something',
        type: 'info',
        duration: '9000',
    }"
    x-on:submit.prevent="$dispatch('snack', { content, type, duration })"
>
...
<label class="c-label" for="duration">
        <span class="c-label__text">Duration:</span>
        <input class="c-input" id="duration" x-model="duration" type="number">
</label>
```

It works in that everything renders and I get no console errors, however it doesn't seem to use duration set within the number input (the notification appears but vanishes pretty quickly). Is this even do-able? Or is this something thats only set or readable from initialisation ?)

If it helps I can create a reduced test case on codepen.

**Note:** Didn't want to paste the entire component for obvious reasons. It's the bread and butter of Alpine.

**Comment by ekwoka** [comment, marked as answer]
```js
this.notification.duration
```

**Reply by aaronstezycki** [reply]
Thanks @ekwoka, feel a little stupid that it's that simple 👍
